### DISCUSSION TITLE
Getting "bundleVersion" from PlayerSettings

### DISCUSSION THREAD

**Discussion by MarshMello0** [nesrak1/AssetsTools.NET#83] [Q&A] [answered]
Hey, I'm trying to see if this library will help me check the version of the game currently installed on a user's machine (without launching the game).

I have found the value I want to check in `globalgamemanagers`. But I can't seem to get a snippet of code to get this value.

Here is the value in Asset View
![image](https://user-images.githubusercontent.com/33008390/189769906-099ccff6-a558-4778-a1d2-3475396c497b.png)

And here is my first attempt at trying to get the value
```cs
var manager = new AssetsManager();
var file = manager.LoadAssetsFile(path, true);
AssetFileInfoEx table = null;
foreach (var currentTable in file.table.GetAssetsOfType((int)AssetClassID.PlayerSettings))
{
    table = currentTable;
    break;
}
var instanceType = manager.GetTypeInstance(file, table);
var baseFields = instanceType.GetBaseField();
var version = baseFields.Get("bundleVersion").GetValue().AsString();
manager.UnloadAll();
```

It throws a null error at the variable `instanceType`. I think I have miss understood how to get the `AssetFileInfoEx` so any guidance would be greatly appreciated.

And the reason for that foreach loop, was I originally tried searching for "PlayerSettings" with `GetAssetInfo("RocketShip");` but it couldn't find it.

Thank you in advance for your help 😁

**Comment by nesrak1** [comment, marked as answer]
Since I assume this is a PC game, and not in a bundle, you will need to load the class database:

```cs
am.LoadClassPackage("classdata.tpk");
am.LoadClassDatabaseFromPackage(inst.file.typeTree.unityVersion);
```

> And the reason for that foreach loop, was I originally tried searching for "PlayerSettings" with GetAssetInfo("RocketShip"); but it couldn't find it.

AssetsView is a bit misleading here. The actual asset isn't named, so AssetsView defaults to the same name as the type. GetAssetInfo only looks for the m_Name field if it exists, so it won't help you. GetAssetsOfType is the right way, although it returns an array so you could just do [0] on it instead of a foreach loop.

Also, PlayerSettings is notorious for changing often and it's possible that you will have read errors or missing fields if you try to read the PlayerSettings asset (it will probably throw an OutOfMemoryException.) In that case, you may want to use the large classdata.tpk file in the releases.

You can also try the new dev branch of AssetsTools.NET which has better versioning support: https://github.com/nesrak1/AssetsTools.NET/tree/upd21-with-inst, but you would need to grab a different kind of tpk from https://nightly.link/AssetRipper/Tpk/workflows/make_tpk/master. The wiki has more info on how to use that new version here: https://github.com/nesrak1/AssetsTools.NET/wiki/Getting-Started:-Assets-file-reading.

**Reply by nesrak1** [reply]
Actually, since you have it open fine in AssetsView you're probably good using the normal version of the library and normal tpk.

**Reply by MarshMello0** [reply]
That explains it.
So I was missing loading a classdata which is provided with AssetView. Those docs for the new version nicely explained what the classdata is for. Turns out I skipped it because I didn't [read the bottom section](https://github.com/nesrak1/AssetsTools.NET#class-database) on the readme and didn't see that file in my games (VTOL VR) directory.

Looking forward to using this in my tool, thanks for your quick reply and for creating this package.
